plenum/test/replica/test_api.py

```diff
@@ -133,4 +148,42 @@ def patched_stateRootHash(self, ledger_id, to_str):
 
     replica.stateRootHash = types.MethodType(patched_stateRootHash, replica)
 
-    assert replica.create3PCBatch(0) is None
\ No newline at end of file
+    assert replica.create3PCBatch(0) is None
+
+
+def test_process_pre_prepare_with_not_final_request(fake_node):
+    fake_node.seqNoDB = FakeSomething(get=lambda req: (None, None))
+    replica = fake_node.replicas[0]
+
+    pp = create_pre_prepare_no_bls(replica.stateRootHash(DOMAIN_LEDGER_ID))
+    replica.nonFinalisedReqs = lambda a: pp.reqIdr
+
+    def reportSuspiciousNodeEx(ex):
+        assert False, ex
+    replica.node.reportSuspiciousNodeEx = reportSuspiciousNodeEx
+
+    def request_propagates(reqs):
+        assert reqs == pp.reqIdr
+    replica.node.request_propagates = request_propagates
+
+    replica.processPrePrepare(pp, replica.primaryName)
+    assert (pp, replica.primaryName, pp.reqIdr) in replica.prePreparesPendingFinReqs
+
+
+def test_process_pre_prepare_with_not_ordered_request(fake_node):
```

Comment:
Should the test be called `test_process_pre_prepare_with_ordered_request`

Reply:
Done